### PR TITLE
Feature/wallet

### PR DESCRIPTION
**Pull request by Nazar865**

### CODE REVIEW

File: Corzbank.Api/Controllers/CardsController.cs [hunk 1]
```diff
@@ -33,6 +33,15 @@ public async Task<IActionResult> GetCards()
             return Ok(result);
         }
 
+        [Route("user/{id}")]
```
Comment (ghost) [inline Corzbank.Api/Controllers/CardsController.cs]: user**S**

File: Corzbank.Helpers/GenerateCard.cs [hunk 2]
```diff
@@ -54,7 +55,7 @@ private static ushort GenerateCvvCode()
                 cvv += rnd.Next(0, 10).ToString();
             }
 
-           return ushort.Parse(cvv);
+            return ushort.Parse(cvv);
```
Comment (ghost) [inline Corzbank.Helpers/GenerateCard.cs:58]: remove space before return.

File: Corzbank.Services/GenericService.cs [hunk 3]
```diff
@@ -115,5 +115,11 @@ public async Task<T> FindByCondition(Expression<Func<T, bool>> expression)
 
 			return result;
         }
+
+		public IEnumerable<T> GetByCondition(Expression<Func<T, bool>> expression)
+		{
+			var result = _dbContext.Set<T>().Where(expression).AsQueryable().ToList();
```
Comment (ghost) [inline Corzbank.Services/GenericService.cs]: Why did you add .AsQueryable and then used .toList?

Comment (ghost) [reply Corzbank.Services/GenericService.cs]: https://stackoverflow.com/a/17996264

File: Corzbank.Services/UserService.cs [hunk 4]
```diff
@@ -60,6 +60,7 @@ public async Task<Token> Login(UserForLoginModel userForLogin)
                 return null;
 
             var tokenForDeleting = await _genericService.FindByCondition(x => x.User.Id == user.Id);
+            if(tokenForDeleting != null)
             await _genericService.Remove(tokenForDeleting);
```
Comment (ghost) [inline Corzbank.Services/UserService.cs]: Add tab here.

**Review by ghost — CHANGES_REQUESTED** [review]

File: Corzbank.Api/Controllers/CardsController.cs (hunk 1 above)
Comment (Nazar865) [reply Corzbank.Api/Controllers/CardsController.cs]: why users if I'll return items for specific user

Comment (ghost) [reply Corzbank.Api/Controllers/CardsController.cs]: REST architecture. collection/specific
https://stackoverflow.com/a/29709867

Comment (ghost) [reply Corzbank.Api/Controllers/CardsController.cs]: user**s**/{id}

Comment (Nazar865) [reply Corzbank.Api/Controllers/CardsController.cs]: sure, thanks)

File: Corzbank.Helpers/GenerateCard.cs (hunk 2 above)
Comment (Nazar865) [reply Corzbank.Helpers/GenerateCard.cs:58]: resolved

File: Corzbank.Services/GenericService.cs (hunk 3 above)
Comment (Nazar865) [reply Corzbank.Services/GenericService.cs]: resolved

File: Corzbank.Services/UserService.cs (hunk 4 above)
Comment (Nazar865) [reply Corzbank.Services/UserService.cs]: resolved

**Review by ghost — APPROVED** [review]
1. **Formatting.**
2. Remove entities from client.
